### ISSUE TITLE
libc vfs_plugin: mkdir fails if directory name ends with slash

### ISSUE THREAD

**Issue by rite** [genodelabs/genode#1330]
I added tests with the directory "testdir3/" to libports/src/test/libc_ffat/main.cc. Now the libc_vfs tests fail:
```
[init -> test-libc_vfs] calling mkdir(dir_name2, 0777) dir_name=testdir2
[init -> test-libc_vfs] mkdir(dir_name2, 0777) succeeded
[init -> test-libc_vfs] calling mkdir(dir_name3, 0777) dir_name=testdir3/
[init -> test-libc_vfs] mkdir(dir_name3, 0777) failed, ret=-1, errno=2
[init] virtual void Genode::Child_policy::exit(int): child "test-libc_vfs" exited with exit value -1
```

**Comment by nfeske** [comment]
A similar problem can be observed when removing a directory recursively in noux:

```
rm -r directory/
```

This results in the following error message:

```
rm: cannot remove `foo/': Operation not permitted
```

However when removing the trailing slash, it works fine.

**Comment by chelmuth** [comment]
To bump this issue I'd like to propose the removal of trailing slashes in the libc before using the VFS. The semantic of the VFS is to lookup the responsible VFS plugin for a request depending on the given path, which is the string in front of the last slash. Are there corner cases I'm missing, e.g., in the context of symbolic links?

**Comment by ehmry** [comment]
I don't think so, I say do the cleanup at the libc, and then use the Path utility to sanitize strings when using the VFS directly.

**Comment by rite** [comment]
:+1:

**Comment by ehmry** [comment]
I've fixed the immediate problem, but there are a probably a few more points on the trailing slash where the libc isn't exactly conformant. There are other situations where libc is supposed to contain logic for this slash but they look like mostly historical artifacts.

**Comment by chelmuth** [comment]
@ehmry that was quick ;-) Is `lstat()` also a candidate?

**Comment by ehmry** [comment]
Is that a riddle? The stuff I read said that a tailing slash can mean dereference to a directory or fail, so who knows what that means for lstat.
http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_12

**Comment by cnuke** [comment]
I would strip the trailing slash everywhere (see http://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xbd_chap04.html#tag_21_04_12).

**Comment by ehmry** [comment]
Yes, that is easy too.

**Comment by chelmuth** [comment]
Sqashed and merged ccbed0dccd09f279b8d25b5f5fb2992ae51c8ad6 and aec6f98c6d88da79dd71be2c6e554bef3c85ed5f.
